src/exo/prec_analysis.py

```diff
@@ -224,6 +246,18 @@ def coerce_e(self, e, btyp):
             assert rhs.type == btyp
 
             return LoopIR.BinOp(e.op, lhs, rhs, btyp, e.srcinfo)
+        elif isinstance(e, LoopIR.Extern):
+            assert e.type == T.R
+            # coerce if T.R
+            args = []
+            for a in e.args:
+                if a.type == T.R:
+                    args.append(self.coerce_e(a, btyp))
```

Comment:
are externs expected to support every available precision on Exo?

Reply:
Yes! But this is coercing the argument when it was T.R (which means the precision was not specified during scheduling)